lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -661,14 +677,17 @@ def generate(cls, _xmodule_instance_args, _entry_id, course_id, task_input, acti
         problem_location = task_input.get('problem_location')
 
         # Compute result table and format it
-        student_data = cls._build_student_data(
+        student_data, student_data_keys = cls._build_student_data(
             user_id=task_input.get('user_id'),
             course_key=course_id,
             usage_key_str=problem_location
         )
 
-        features = ['username', 'title', 'location', 'block_key', 'state']
-        header, rows = format_dictlist(student_data, features)
+        for data in student_data:
+            for key in student_data_keys:
```

Comment:
what about the oneliner? as it is only used for setting default values.
`[data.setdefault(key, '') for key in student_data_keys]`

I am open to this change. its up to you.

Reply:
I think that would be a bit less clear, would prefer to keep it the way it is. No strong feeling either way though!